lib/iser.c

```diff
@@ -201,7 +203,17 @@ iser_free_iser_conn_res(struct iser_conn *iser_conn, bool destroy)
 	if (destroy) {
 
 		if (iser_conn->cmthread) {
-			pthread_cancel(iser_conn->cmthread);
+			int retry = ISER_CM_THREAD_WAIT;
+
+			iser_conn->cmthread_should_run = 0;
+			while ((retry-- > 0) && (iser_conn->conn_state == CONN_ESTABLISHED)) {
+				usleep(1);
+			}
+
+			if (iser_conn->conn_state == CONN_ESTABLISHED) {
+				ISCSI_LOG(iscsi, 2, "iser_free_iser_conn_res: try to kill cm thread after destroy qp %dms", ISER_CM_THREAD_WAIT);
+				pthread_cancel(iser_conn->cmthread);
```

Comment:
The commit message explains why calling pthread_cancel() can cause trouble. So please remove all pthread_cancel() calls.